catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/UpdateOperations.java

```diff
@@ -647,4 +670,45 @@ private void logFailedQueryInfo(UpdateRequest updateRequest, QueryResponse query
                             .collect(Collectors.joining(", ", "[", "]")));
         }
     }
+
+    private UpdateStorageRequest applyAttributeOverrides(UpdateStorageRequest updateStorageRequest,
+            Map<String, Metacard> metacardMap) {
+        for (ContentItem contentItem : updateStorageRequest.getContentItems()) {
+            Metacard metacard = metacardMap.get(contentItem.getId());
+            MetacardType metacardType = metacard.getMetacardType();
+
+            if (contentItem.getMetacard() != null) {
+                Metacard overrideMetacard = contentItem.getMetacard();
+                MetacardType overrideMetacardType = contentItem.getMetacard()
+                        .getMetacardType();
+
+                if (overrideMetacardType.equals(BasicTypes.BASIC_METACARD)) {
```

Comment:
You knew this comment was coming but I think you can extract this into a class that extends MetacardImpl or create a factory that merges overrides.  That class can just live in the standard framework project.  Then both operations can use this common logic.